src/D_functions_and_procedures.sql

```diff
@@ -362,4 +362,192 @@ $$ LANGUAGE plpgsql;
 
 -- D12 TEST
 SELECT * FROM best_match_titles(ARRAY['apple', 'mads', 'mikkelsen']);
-SELECT title FROM release WHERE media_id=47460;
\ No newline at end of file
+SELECT title FROM release WHERE media_id=47460;
+
+
+
+-------------------D9 SIMILAR MOVIES SEARCH-------------------------
+--------------------------------------------------------------------
+
+--FINAL FUNCTION SIMILAR MOVIES
+--THIS COMBINES OTHER FUNCTIONS WHICH ARE DECLARED LOWER IN THIS FILE
+CREATE OR REPLACE FUNCTION get_similar_movies(input_media_id INTEGER)
+RETURNS TABLE (
+	media_id INTEGER,
+	count INTEGER
+) AS $$
+
+DECLARE
+	input_media RECORD;
+BEGIN
+
+  --get title id and country of input_media_id
+	SELECT DISTINCT m.media_id, r.title, mpc.country_id INTO input_media
+	FROM media m
+	JOIN release r USING (media_id)
+	LEFT JOIN media_production_country mpc USING(media_id)
+	WHERE m.media_id = input_media_id;
+	
+	RETURN query
+	SELECT com_res.media_id, SUM(com_res.count)::INTEGER AS total_count
+	FROM (
+		SELECT *  FROM get_count_of_movies_with_same_actors(input_media_id) 
+		UNION ALL
+		SELECT *  FROM find_movies_from_the_same_country(input_media.country_id)
+		UNION ALL
+		SELECT *  FROM find_movie_similar_titles(input_media.title,3)
+		UNION ALL
+		SELECT *  FROM find_movies_with_similar_crew(input_media_id)
+	) AS com_res
+	GROUP BY com_res.media_id
+	ORDER BY total_count DESC;
+END;
+$$ LANGUAGE plpgsql;
+
+-------------------------------------------------------
+---------get_count_of_movies_with_same_actors----------
+	
+CREATE OR REPLACE FUNCTION get_count_of_movies_with_same_actors(input_media_id INTEGER)
+RETURNS TABLE(
+	media_id INTEGER,
+	count INTEGER
+) AS $$
+DECLARE
+	actor INTEGER;
+	actors INTEGER[];
+BEGIN
+	SELECT ARRAY_AGG(DISTINCT cm.person_id) INTO actors
+	FROM cast_member cm
+	WHERE cm.media_id = input_media_id;
+	
+	CREATE TEMPORARY TABLE similar_cast (
+	media_id INTEGER,
+	"count" INTEGER
+	) ON COMMIT DROP;
+	
+	FOREACH actor IN ARRAY COALESCE(actors, '{}')
+	LOOP
+		INSERT INTO similar_cast(media_id, "count")
+		SELECT DISTINCT cm.media_id, 1
+		FROM cast_member cm
+		WHERE cm.person_id = actor
+		GROUP BY cm.media_id;
+	END LOOP;
+	
+	RETURN query
+	SELECT s.media_id, count(s.media_id)::INTEGER
+	FROM similar_cast s
+	GROUP BY s.media_id;
```

Comment:
I maybe think it whould be nice to have it ordered by the count in DESC, when I guess you could see the most relevant media.

And also this is needs to be applied on all.